### PR TITLE
feat: add upgrade banner for SLA feature

### PR DESCRIPTION
**Pull request by scmmishra**
image

Add upgrade banner to SLA feature

### CODE REVIEW

File: enterprise/app/services/enterprise/billing/handle_stripe_event_service.rb [hunk 1]
```diff
@@ -61,7 +61,7 @@ def ensure_event_context(event)
   end
 
   def features_to_update
-    %w[help_center campaigns team_management channel_twitter channel_facebook channel_email]
+    %w[help_center campaigns team_management sla channel_twitter channel_facebook channel_email]
```
Comment (scmmishra) [inline enterprise/app/services/enterprise/billing/handle_stripe_event_service.rb:64]: @pranavrajs @sojan-official @vishnu-narayanan seems like we enable all the features for any paid plan, even if it's available on the higher tier, should I fix it?

**Comment by vishnu-narayanan** [comment]
@scmmishra Who all this CTA would be shown?

1. Free Account without `sla-feature` enabled 
2. Free Account with `sla-feature` enabled

How would this be shown for normal-user/account-admin/super admin users for cases 1 and 2?

**Comment by scmmishra** [comment]
> @scmmishra To whom should this CTA be shown?
> 
> 1. Free Account without `sla-feature` enabled
> 2. Free Account with `sla-feature` enabled

At the moment, this is shown to all accounts, regardless of the `sla-feature` being enabled or not. 

> How would this be shown for normal-user/account-admin/super admin users for cases 1 and 2?

Missed handling this for non-admin users. Will update that

**Comment by scmmishra** [comment]
@vishnu-narayanan handled the case for non-admin users

CleanShot 2024-04-15 at 14 36 32@2x

File: app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAEmptyState.vue [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<script setup>
+import BaseEmptyState from './BaseEmptyState.vue';
+import { defineEmits } from 'vue';
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAEmptyState.vue:3]: We don't need to import `defineEmits`.
```suggestion
```

File: app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAPaywall.vue [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    name: 'billing_settings_index',
+    params: { accountId },
+  });
+}
+</script>
+
+<template>
+  <base-empty-state>
+    <div
+      class="flex flex-col items-center max-w-md px-6 pt-6 pb-3 bg-white border shadow rounded-xl border-slate-100"
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAPaywall.vue:32]: The border and background color are missing for the dark mode.

File: app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAPaywall.vue [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    >
+      <div class="flex items-center w-full gap-2">
+        <span
+          class="flex items-center justify-center w-6 h-6 rounded-full bg-woot-100"
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAPaywall.vue:36]: The background color is missing for the dark mode.

File: app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAPaywall.vue [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        </woot-button>
+        <p class="mt-2 text-xs tracking-tighter text-center">
+          {{ $t('SLA.PAYWALL.CANCEL_ANYTIME') }}
+        </p>
+      </template>
+      <template v-else>
+        <p class="text-sm font-normal tracking-[0.5%]">
+          {{ $t('SLA.PAYWALL.ASK_ADMIN') }}
+        </p>
+      </template>
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAPaywall.vue:69]: Please update the text color for dark mode

**Review by iamsivin — COMMENTED** [review]
@scmmishra I have added some comments. Could you please take a look?

File: enterprise/app/services/enterprise/billing/handle_stripe_event_service.rb (hunk 1 above)
Comment (sojan-official) [reply enterprise/app/services/enterprise/billing/handle_stripe_event_service.rb:64]: Also need to verify the cases for the community edition and enterprise edition. 

- community edition:  we shouldn't show this page on the community edition ( or show ? )
- enterprise edition
   - paid with chatwoot meta disabled : should not show
   - non paid : show

File: app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAPaywall.vue (hunk 4 above)
Comment (scmmishra) [reply app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAPaywall.vue:36]: @iamsivin handled this in https://github.com/chatwoot/chatwoot/pull/9240, will port it back here once the PR is merged

File: app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAPaywall.vue (hunk 5 above)
Comment (scmmishra) [reply app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAPaywall.vue:69]: @iamsivin handled this in https://github.com/chatwoot/chatwoot/pull/9240, will port it back here once the PR is merged

**Comment by scmmishra** [comment]
Already handled in a separate PR

**Pull request closed without merging**